### DISCUSSION TITLE
[compilcast21] Unsupported behaviour type - error in BentoniteMassin2017.mfront law

### DISCUSSION THREAD

**Discussion by ra1106** [thelfer/tfel#372] [Q&A] [answered]
Hello,
I work with Mfront laws in Cast3m as Iwan. I have seen that you have a hypoplastic law called BentoniteMassin2017 from [https://soilmodels.com/hypoclay/.](https://soilmodels.com/hypoclay/) I downloaded the .mfront folder from the https://github.com/thelfer/MFrontGallery . I tried to compile it as I normally do in Cast3m with "compilcast21 BentoniteMassin2017.mfront" and my error was the following: Unsupported behaviour type. It created the src and include folders but not the Castem file. Do you know the reason for this error?
I thank you in advance for your time.
Cordially,

**Comment by thelfer** [comment, marked as answer]
Hi @ra1106,

The bentonite behaviour is a generalized behaviour which computes the stress **and** the saturation as a function of the strain and the suction.

Such behaviour is not supported by `Cast3M` at the moment.

If the evolution of the suction is known/imposed, we could try to create a wrapper for this behaviour to be used in a classical mechanical resolution....

Thomas

**Reply by ra1106** [reply]
Hello, 

Thank you for your quick response. I was actually looking for the "simpler" version of the model. The one found in https://soilmodels.com/download/plaxis-umat-hypoplas-zip/ available in UMAT. Mr Helfer, do you know if an interface like mfront is available to introduce an already existing UMAT law in Cast3m ?
Thank you for your answer. 
Cordially,

**Reply by thelfer** [reply]
The question is quite tricky. There is an UMAT interface in Cast3M but for many reasons it is sightly different for Abaqus UMAT.
You will have to wrap the abaqus UMAT.

 One solution is to follow this tutorial:
<https://thelfer.github.io/MFrontGallery/web/SmallStrainUmatWrapper.html>

**Reply by ra1106** [reply]
Oh! I expected to find a simple user interface like mfront. 
Have a nice day. Thank you.

**Reply by thelfer** [reply]
In a sense, the `Cast3M` interface is simple. But you'll have to take care of details that `MFront` handles automatically.
